epix/io.py

```diff
@@ -196,6 +197,139 @@ def _get_ttree(directory, file_name):
+
+
+    if outer_cylinder:
+        cut_string = (f'(r < {outer_cylinder["max_r"]})'
+                      f' & ((z >= {outer_cylinder["min_z"] * 10}) & (z < {outer_cylinder["max_z"] * 10}))')
+        instr_df = instr_df.query(cut_string)
+    
+    interactions = awkwardify_df(instr_df)
+
+    if np.any(interactions['ed'] < 0):
+        warnings.warn('At least one of the energy deposits is negative!')
+    
+    # Removing all events with zero energy deposit
+    m = interactions['ed'] > 0
+    if cut_by_eventid:
+        # ufunc does not work here...
+        m2 = (interactions['evtid'] >= start) & (interactions['evtid'] < stop)
+        m = m & m2
+    interactions = interactions[m]
+
+    # Removing all events with no interactions:
+    m = ak.num(interactions['ed']) > 0
+    interactions = interactions[m]
+
+    return interactions, n_simulated_events
+
+
+def awkwardify_df(df):
+    
+    _, evt_offsets = np.unique(df["eventid"], return_counts = True)
+    
+    dictionary = {"x": reshape_awkward(df["x"].values , evt_offsets),
+                  "y": reshape_awkward(df["y"].values , evt_offsets),
+                  "z": reshape_awkward(df["z"].values , evt_offsets),
+                  "r": reshape_awkward(df["r"].values , evt_offsets),
+                  "t": reshape_awkward(df["t"].values , evt_offsets),
+                  "ed": reshape_awkward(df["ed"].values , evt_offsets),
+                  "type":reshape_awkward(np.array(df["type"], dtype="<U5") , evt_offsets),
+                  "trackid": reshape_awkward(df["trackid"].values , evt_offsets),
+                  "parenttype": reshape_awkward(np.array(df["parenttype"], dtype="<U5") , evt_offsets),
+                  "parentid": reshape_awkward(df["parentid"].values , evt_offsets),
+                  "creaproc": reshape_awkward(np.array(df["creaproc"], dtype="<U5") , evt_offsets),
+                  "edproc": reshape_awkward(np.array(df["edproc"], dtype="<U5") , evt_offsets),
+                  "evtid": reshape_awkward(df["eventid"].values , evt_offsets),
+                 }
+    
+    return ak.Array(dictionary)
+
+#In principle these function are already in common.py 
+#i needed to change res.real to res.append but this is not numba compatible
+#find a good solution in the PR!
+def reshape_awkward(array, offset):
+    """
+    Function which reshapes array according to a list of offsets. Only
+    works for a single jagged layer.
+    Args:
+        array: Flatt array which should be jagged.
+        offset: Length of subintervals
+    Returns:
+        res: awkward1.ArrayBuilder object.
+    """
+    res = ak.ArrayBuilder()
+    _reshape_awkward(array, offset, res)
+    return res.snapshot()
+
+def _reshape_awkward(array, offsets, res):
+    start = 0
+    end = 0
+    for o in offsets:
+        end += o
+        res.begin_list()
+        for value in array[start:end]:
+            res.append(value)
+        res.end_list()
+        start = end
```

Comment:
These lines are basically a copy of https://github.com/XENONnT/epix/blob/master/epix/common.py#L6-L34 but `res.real(value)` was changed to `res.append(value)` to allow lists/arrays of strings as input.

Reply:
Why do you have to copy both in that case? You could modify `reshape_awkward` and call the correct version of  `_reshape_awkward`depending on the users input.